Source/HTTP/XMLHttp/xmlhttp_http_task.cpp

```diff
@@ -131,12 +130,12 @@ void xmlhttp_http_task::perform_async(
             return;
         }
 
-        if (requestBodyBytes > 0 && requestBody != nullptr)
+        if (requestBodySize > 0 && requestBodyReadFunction != nullptr)
```

Comment:
is it ever valid to have a body size > 0 but no read function?

Reply:
No, we should always have a read and write function